demo/api/src/config/environment-variables.ts

```diff
@@ -122,6 +122,28 @@ export class EnvironmentVariables {
     @IsString()
     S3_BUCKET: string;
 
+    @IsString()
+    MAILER_HOST: string;
+
+    @Type(() => Number)
+    @IsInt()
+    MAILER_PORT: number;
+
+    @IsString()
+    MAILER_DEFAULT_FROM: string;
```

Comment:
is this really configured differently per environment?

Reply:
> and in our cases it's stored in our vaults and it doesn't matter

Every extra env var needs to bet set/handled at multiple places. That does matter imho.

Possible alternatives:
- hardcode in the AppModule (where MailerModule is configured)
- add to comet-config.json (and access that in AppModule - where MailerModule is configured)
    - (comet-config.json is for config values that won't need different values per deployment)

...and if an application really needs it, it can always add it.


> but sometimes it's nice to know if the received mail was sent from prod or non-prod

If that is an requirement (and actually used) then it should be an env var

Reply:
> But I still think we still will provide those values as env-vars for our projects, so I didn't change the docs.

if you think so, why do you change it? Demo should be as close to real applications as possible.


MAILER_HOST, MAILER_PORT and MAILER_SEND_ALL_MAILS_TO will be different per deployment. Tor those you should keep the env var. My comment was referring to MAILER_DEFAULT_FROM only.

Reply:
> if you think so, why do you change it? Demo should be as close to real applications as possible.

ah, I wasn't sure about that. I thought demo can be slightly different.

For me this is hard to guess. I assumed MAILER_HOST and MAILER_PORT will be the same. Because the mail-server wont probably be part of deployment but provided as a central service (smtp server for instance) used by all deployments. 

Maybe I should just revert the commit, do you agree?

Reply:
> ah, I wasn't sure about that. I thought demo can be slightly different.

if there are reasons for beeing different, yes. But we usually try to avoid them.

> I assumed MAILER_HOST and MAILER_PORT will be the same

no: local development uses different values as any deployed environmnt

> Maybe I should just revert the commit, do you agree?

partially; I suggest to remove `MAILER_DEFAULT_FROM` env var and hardcode that in AppModule.